src/main/java/beam/utils/logging/LogAggregatorAppender.java

```diff
@@ -0,0 +1,100 @@
+package beam.utils.logging;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.ConsoleAppender;
+import org.apache.commons.math3.ml.clustering.Cluster;
+import org.apache.commons.math3.ml.clustering.DBSCANClusterer;
+import org.apache.commons.text.similarity.LongestCommonSubsequence;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LogAggregatorAppender extends ConsoleAppender<ILoggingEvent> {
+
+    private List<String> allMessages = new LinkedList<>();
```

Comment:
This doesn't seem like it would be good for memory - storing all messages until the end afaict?